engines/gob/inter_v7.cpp

```diff
@@ -391,33 +698,87 @@ void Inter_v7::o7_playVmdOrMusic() {
 	}
 
 }
-void Inter_v7::o7_draw0x89() {
+void Inter_v7::o7_setActiveCD() {
 	Common::String str0 = _vm->_game->_script->evalString();
 	Common::String str1 = _vm->_game->_script->evalString();
 
-	warning("Addy Stub Draw 0x89: \"%s\", \"%s\"", str0.c_str(), str1.c_str());
-
-	if (findFile(str0).empty()) {
-		storeValue(0);
-		return;
+	Common::ArchiveMemberList files;
+	SearchMan.listMatchingMembers(files, str0);
+	Common::String savedCDpath = _currentCDPath;
+
+	for (Common::ArchiveMemberPtr file : files) {
+		auto *node = dynamic_cast<Common::FSNode *>(file.get());
+		if (node != nullptr &&
+			setCurrentCDPath(node->getParent())) {
+			debugC(5, kDebugFileIO, "o7_setActiveCD: %s -> %s", savedCDpath.c_str(),  _currentCDPath.c_str());
+			storeValue(1);
+			return;
+		}
 	}
 
-	storeValue(1);
+	storeValue(0);
 }
 
 void Inter_v7::o7_findFile() {
-	Common::String file = findFile(getFile(_vm->_game->_script->evalString()));
+	Common::String file_pattern = getFile(_vm->_game->_script->evalString());
+	Common::ArchiveMemberList files;
 
-	storeString(file.c_str());
-	storeValue(file.empty() ? 0 : 1);
+	SearchMan.listMatchingMembers(files, file_pattern);
+	Common::ArchiveMemberList filesWithoutDuplicates;
+	for (Common::ArchiveMemberPtr file : files)
```

Comment:
Formatting for this whole block is off: all opening curly braces must stick to the previous line